Core/src/org/sleuthkit/autopsy/filesearch/FileSearchPanel.java

```diff
@@ -170,54 +177,77 @@ private boolean isValidSearch() {
         try {
             if (this.isValidSearch()) {
-                String title = NbBundle.getMessage(this.getClass(), "FileSearchPanel.search.results.title", ++resultWindowCount);
-                String pathText = NbBundle.getMessage(this.getClass(), "FileSearchPanel.search.results.pathText");
-
+                // change the cursor to "waiting cursor" for this operation
+                WindowManager.getDefault().getMainWindow().setCursor(Cursor.getPredefinedCursor(Cursor.WAIT_CURSOR));
                 // try to get the number of matches first
                 Case currentCase = Case.getCurrentCaseThrows(); // get the most updated case
-                long totalMatches = 0;
-                List<AbstractFile> contentList = null;
-                try {
-                    SleuthkitCase tskDb = currentCase.getSleuthkitCase();
-                    contentList = tskDb.findAllFilesWhere(this.getQuery());
-
-                } catch (TskCoreException ex) {
-                    Logger logger = Logger.getLogger(this.getClass().getName());
-                    logger.log(Level.WARNING, "Error while trying to get the number of matches.", ex); //NON-NLS
-                }
-
-                if (contentList == null) {
-                    contentList = Collections.<AbstractFile>emptyList();
-                }
-
-                SearchNode sn = new SearchNode(contentList);
-                TableFilterNode tableFilterNode = new TableFilterNode(sn, true, sn.getName());
-                final TopComponent searchResultWin;
-                if (contentList.isEmpty()) {
-                    Node emptyNode = new TableFilterNode(new EmptyNode(Bundle.FileSearchPanel_emptyNode_display_text()), true);
-                    searchResultWin = DataResultTopComponent.createInstance(title, pathText,
-                        emptyNode, 0);
-                } else {
-                    searchResultWin = DataResultTopComponent.createInstance(title, pathText,
-                        tableFilterNode, contentList.size());
-                }
-                searchResultWin.requestActive(); // make it the active top component
-
-                /**
-                 * If total matches more than 1000, pop up a dialog box that say
-                 * the performance maybe be slow and to increase the
-                 * performance, tell the users to refine their search.
-                 */
-                if (totalMatches > 10000) {
-                    // show info
-                    String msg = NbBundle.getMessage(this.getClass(), "FileSearchPanel.search.results.msg", totalMatches);
-                    String details = NbBundle.getMessage(this.getClass(), "FileSearchPanel.search.results.details");
-                    MessageNotifyUtil.Notify.info(msg, details);
-                }
+                searchWorker = new SwingWorker<Void, Void>() {
+                    List<AbstractFile> contentList = null;
+                    TableFilterNode tableFilterNode = null;
```

Comment:
I would probably make a little Result class for contentList and tableFilterNode and have doInBackground return that class.